### PR TITLE
Documentation/improve gharial documentation

### PR DESCRIPTION
**Pull request by mchacki**
Improves the Graph HTTP API documentation.

Should now cover all http input parameters and all possible outcomes, including error codes and examples.

### CODE REVIEW

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md [hunk 1]
```diff
@@ -1,30 +1,101 @@
...
 @RESTRETURNCODE{201}
-Returned if the edge collection could be added successfully and
-waitForSync is true.
+Is returned if the collection could be created and waitForSync is enabled
+for the `_graphs` collection, or given in the request.
+The response body contains the graph configuration that has been stored.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md]: Should this be here? I see this multiple times. If it is correct, then we should add a dot (.) after `(false)`.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md [hunk 2]
```diff
@@ -1,30 +1,101 @@
...
 @RESTRETURNCODE{202}
-Returned if the edge collection could be added successfully and
-waitForSync is false.
+Is returned if the collection could be created and waitForSync is disabled
+for the `_graphs` collection, or given in the request.
+The response body contains the graph configuration that has been stored.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md]: Should this be here? I see this multiple times. If it is correct, then we should add a dot (.) after (false).

File: Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md [hunk 3]
```diff
@@ -8,43 +8,107 @@ The creation of a graph requires the name of the graph and a
...
+
+  1. `Administrate` access on the Database.
+  2. `Read Only` access on every collection used within this graph.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md]: dot missing at the end.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md [hunk 4]
```diff
@@ -53,6 +117,19 @@ edge definition with a the same
 [edge collection](../../Manual/Appendix/Glossary.html#edge-collection) but a
 different signature used in any other graph.
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md]: dot missing at the end.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md [hunk 5]
```diff
@@ -40,22 +40,113 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+
+@RESTREPLYBODY{errorMessage,string,required,}
+A message created for this error.
 
 @RESTRETURNCODE{412}
-Returned if if-match header is given, but the documents revision is different.
+Returned if if-match header is given, but the stored documents revision is different.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md [hunk 6]
```diff
@@ -40,22 +40,113 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
 
 @RESTRETURNCODE{404}
-Returned if no graph with this name, no vertex collection or no vertex with this id could be found.
+Returned in the following cases:
+* No graph with this name could be found.
+* This collection is not part of the graph.
+* The vertex to replace does not exist.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md [hunk 7]
```diff
@@ -40,22 +40,113 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_replace_http_examples.md]: dot missing?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_modify_http_examples.md [hunk 8]
```diff
@@ -40,22 +43,113 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_modify_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md [hunk 9]
```diff
@@ -1,40 +1,129 @@
...
+
+@RESTREPLYBODY{errorMessage,string,required,}
+A message created for this error.
 
 @RESTRETURNCODE{412}
-Returned if if-match header is given, but the documents revision is different.
+Returned if if-match header is given, but the stored documents revision is different.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md [hunk 10]
```diff
@@ -1,40 +1,129 @@
...
+
+  1. `Read Only` access on the Database.
+  2. `Read Only` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md]: dot missing

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md [hunk 11]
```diff
@@ -1,40 +1,129 @@
...
+currently stored vertex still has this revision value.
+So there was no update between the last time the vertex
+was fetched by the caller.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_get_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_delete_http_examples.md [hunk 12]
```diff
@@ -43,14 +38,93 @@ you can supply the Etag in an attribute rev in the URL.
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_delete_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_create_http_examples.md [hunk 13]
```diff
@@ -1,38 +1,108 @@
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_create_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_remove_http_examples.md [hunk 14]
```diff
@@ -23,21 +25,85 @@ Collection will only be dropped if it is not used in other graphs.
...
+
 @RESTRETURNCODE{202}
 Returned if the request was successful but waitForSync is false.
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_remove_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_remove_http_examples.md [hunk 15]
```diff
@@ -23,21 +25,85 @@ Collection will only be dropped if it is not used in other graphs.
...
+@RESTRETURNCODE{200}
 Returned if the vertex collection was removed from the graph successfully
 and waitForSync is true.
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_remove_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md [hunk 16]
```diff
@@ -1,30 +1,101 @@
...
+
+  1. `Administrate` access on the Database.
+  2. `Read Only` access on every collection used within this graph.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md [hunk 17]
```diff
@@ -1,30 +1,101 @@
...
+Is returned if the collection could be created and waitForSync is disabled
+for the `_graphs` collection, or given in the request.
+The response body contains the graph configuration that has been stored.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_vertex_collection_add_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_get_http_examples.md [hunk 18]
```diff
@@ -2,25 +2,50 @@
...
-Returned if the graph could be found.
+Returns the graph if it could be found.
+The result will have the following format:
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_get_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md [hunk 19]
```diff
@@ -39,22 +40,120 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md [hunk 20]
```diff
@@ -39,22 +40,120 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code.
+
+@RESTREPLYBODY{edge,object,required,edge_representation}
+The internal attributes for the edge
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md [hunk 21]
```diff
@@ -39,22 +40,120 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_replace_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 22]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
+A message created for this error.
+
+@RESTRETURNCODE{412}
+Returned if if-match header is given, but the stored documents revision is different.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 23]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 24]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+Will only be present if returnOld is true.
+
+@RESTRETURNCODE{403}
+Returned if your user has insufficient rights.
+In order to update edges in the graph  you at least need to have the following privileges:
+
+  1. `Read Only` access on the Database.
+  2. `Write` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 25]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 26]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+
+@RESTREPLYBODY{old,object,optional,edge_representation}
+The complete overwritten edge document.
+Includes all attributes stored before this operation.
+Will only be present if returnOld is true.
 
 @RESTRETURNCODE{202}
 Returned if the request was successful but waitForSync is false.
 
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 27]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 28]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
 
 @RESTRETURNCODE{200}
-Returned if the edge could be updated.
+Returned if the edge could be updated, and waitForSync is false.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
+It is false in this response.
+
+@RESTREPLYBODY{code,integer,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 29]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
...
+@RESTALLBODYPARAM{edge,object,required}
+The body has to contain a JSON object containing exactly the attributes that should be overwritten, all other attributes remain unchanged.
 
 @RESTRETURNCODES
 
 @RESTRETURNCODE{200}
-Returned if the edge could be updated.
+Returned if the edge could be updated, and waitForSync is false.
+
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md [hunk 30]
```diff
@@ -24,31 +23,136 @@ The *_key* attribute of the vertex.
 Define if the request should wait until synced to disk.
 
 @RESTQUERYPARAM{keepNull,boolean,optional}
-Define if values set to null should be stored. By default the key is not removed from the document.
+Define if values set to null should be stored.
+By default (true) the given documents attribute(s) will be set to null.
+If this parameter is false the attribute(s) will instead be delete from the
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_modify_http_examples.md]: typo: delete

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md [hunk 31]
```diff
@@ -25,16 +32,98 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+Flag if there was an error (true) or not (false)
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code.
+
+@RESTREPLYBODY{errorNum,integer,required,}
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md]: is the last "," intended? 

also see + 20 rows above (same)

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md [hunk 32]
```diff
@@ -25,16 +32,98 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+
+  1. `Read Only` access on the Database.
+  2. `Read Only` access on the given collection.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md [hunk 33]
```diff
@@ -25,16 +32,98 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+currently stored edge still has this revision value.
+So there was no update between the last time the edge
+was fetched by the caller.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md [hunk 34]
```diff
@@ -25,16 +32,98 @@ If the "If-Match" header is given, then it must contain exactly one Etag. The do
...
+Returned if the if-none-match header is given and the
+currently stored edge still has this revision value.
+So there was no update between the last time the edge
+was fetched by the caller.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_get_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_delete_http_examples.md [hunk 35]
```diff
@@ -39,14 +38,93 @@ you can supply the Etag in an attribute rev in the URL.
...
+Flag if there was an error (true) or not (false)
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code.
+
+@RESTREPLYBODY{errorNum,integer,required,}
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_delete_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_delete_http_examples.md [hunk 36]
```diff
@@ -39,14 +38,93 @@ you can supply the Etag in an attribute rev in the URL.
 @RESTRETURNCODE{200}
 Returned if the edge could be removed.
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_delete_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md [hunk 37]
```diff
@@ -29,15 +29,80 @@ One or many vertex collections that can contain target vertices.
...
+Flag if there was an error (true) or not (false)
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code.
+
+@RESTREPLYBODY{errorNum,integer,required,}
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md [hunk 38]
```diff
@@ -29,15 +29,80 @@ One or many vertex collections that can contain target vertices.
...
+A message created for this error.
+
+@RESTRETURNCODE{403}
+Returned if your user has insufficient rights.
+In order to drop a vertex you at least need to have the following privileges:
+  1. `Administrate` access on the Database.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md [hunk 39]
```diff
@@ -29,15 +29,80 @@ One or many vertex collections that can contain target vertices.
 @RESTRETURNCODE{201}
 Returned if the request was successful and waitForSync is true.
 
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_modify_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_add_http_examples.md [hunk 40]
```diff
@@ -32,18 +32,86 @@ One or many vertex collections that can contain target vertices.
...
+
+@RESTREPLYBODY{graph,object,required,graph_representation}
+The information about the modified graph.
 
 @RESTRETURNCODE{202}
 Returned if the definition could be added successfully and
 waitForSync is disabled for the `_graphs` collection.
+The response body contains the graph configuration that has been stored.
+
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_add_http_examples.md]: is the last "," intended?

Comment (hkernbach) [reply Documentation/DocuBlocks/Rest/Graph/general_graph_edge_definition_add_http_examples.md]: also see lines above and under.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_create_http_examples.md [hunk 41]
```diff
@@ -1,45 +1,134 @@
...
+
 @RESTRETURNCODE{404}
-Returned if no graph with this name, no edge collection or no edge with this id could be found.
+Returned in any of the following cases:
+* no graph with this name could be found.
+* this edge collection is not part of the graph.
+* either `_from` or `_to` vertex does not exist.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false)
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_create_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_edge_create_http_examples.md [hunk 42]
```diff
@@ -1,45 +1,134 @@
...
+@RESTRETURNCODE{400}
+Returned if the input document is invalid.
+This can for instance be the case if `_from` or `_to` is mising.
+
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_edge_create_http_examples.md]: dot missing.

File: Documentation/DocuBlocks/Rest/Graph/general_graph_drop_http_examples.md [hunk 43]
```diff
@@ -21,15 +23,46 @@ dropped if they are not used in other graphs.
...
+@RESTREPLYBODY{errorNum,integer,required,}
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
+A message created for this error.
 
 @RESTRETURNCODE{404}
 Returned if no graph with this name could be found.
 
+@RESTREPLYBODY{error,boolean,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_drop_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md [hunk 44]
```diff
@@ -53,6 +117,19 @@ edge definition with a the same
...
+
+@RESTREPLYBODY{errorNum,integer,required,}
+ArangoDB error number for the error that occured.
+
+@RESTREPLYBODY{errorMessage,string,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md [hunk 45]
```diff
@@ -53,6 +117,19 @@ edge definition with a the same
...
 different signature used in any other graph.
 
+@RESTREPLYBODY{error,boolean,required,}
+Flag if there was an error (true) or not (false).
+It is true in this response.
+
+@RESTREPLYBODY{code,integer,required,}
+The response code
+
+@RESTREPLYBODY{errorNum,integer,required,}
```
Comment (hkernbach) [inline Documentation/DocuBlocks/Rest/Graph/general_graph_create_http_examples.md]: is the last "," intended?

File: Documentation/DocuBlocks/Rest/Graph/general_graph_drop_http_examples.md (hunk 43 above)
Comment (mchacki) [reply Documentation/DocuBlocks/Rest/Graph/general_graph_drop_http_examples.md]: it is, the script requires exactly 4 arguments, but the last one is allowed to be empty. ...

Comment (hkernbach) [reply Documentation/DocuBlocks/Rest/Graph/general_graph_drop_http_examples.md]: Ok, thanks. Was expecting something like "optional" for that.

**Comment by hkernbach** [comment]
Latest changes LGTM.

**Pull request merged**